src/coreclr/debug/di/process.cpp

```diff
@@ -8717,19 +8688,23 @@ CordbAppDomain * CordbProcess::LookupOrCreateAppDomain(VMPTR_AppDomain vmAppDoma
     return CacheAppDomain(vmAppDomain);
 }
 
-CordbAppDomain * CordbProcess::GetSharedAppDomain()
+CordbAppDomain * CordbProcess::GetAppDomain()
 {
-    if (m_sharedAppDomain == NULL)
+    // Return the one and only app domain
+    HASHFIND find;
+    CordbAppDomain* appDomain = m_appDomains.FindFirst(&find);
```

Comment:
Any reason to walk the hashtable and then do the `GetCurrentAppDomain` dance?  Just for faster lookup?

Reply:
Yeah, just faster lookup / avoiding the actual dac call if we don't need it.